x-pack/plugins/reporting/server/lib/event_logger/adapter.ts

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import deepMerge from 'deepmerge';
+import { LogMeta } from 'src/core/server';
+import { LevelLogger } from '../level_logger';
+import { IReportingEventLogger } from './logger';
+
+/** @internal */
+export class EcsLogAdapter implements IReportingEventLogger {
+  start?: Date;
+  end?: Date;
+
+  /**
+   * This class provides a logging system to Reporting code, using a shape similar to the EventLog service.
+   * The logging action causes ECS data with Reporting metrics sent to DEBUG logs.
+   *
+   * @param {LevelLogger} logger - Reporting's wrapper of the core logger
+   * @param {Partial<LogMeta>} properties - initial ECS data with template for Reporting metrics
+   */
+  constructor(private logger: LevelLogger, private properties: Partial<LogMeta>) {}
+
+  logEvent(message: string, properties: LogMeta) {
+    if (this.start && !this.end) {
+      this.end = new Date(Date.now());
+    }
+
+    // add the derived properties for timing between "start" and "complete" logging calls
+    const newProperties: LogMeta = deepMerge(this.properties, {
+      event: {
+        duration: this.end && this.start ? this.end.valueOf() - this.start.valueOf() : undefined,
```

Comment:
ECS event.duration [should be in nanoseconds](https://www.elastic.co/guide/en/ecs/master/ecs-event.html#field-event-duration), so you'll need to do some additional math to convert this from milliseconds if you want to keep this ECS-compliant (typescript obviously can't enforce this, which is why it isn't complaining)